src/compiler/crystal/codegen/primitives.cr

```diff
@@ -668,6 +683,16 @@ class Crystal::CodeGenVisitor
   end
 
   def codegen_convert(from_type : IntegerType, to_type : FloatType, arg, *, checked : Bool)
+    # TODO remove conditional after 0.28
+    if @program.has_flag?("preview_overflow")
+      if checked
+        if from_type.kind == :u128 && to_type.kind == :f32
```

Comment:
this check is done twice

Reply:
yes, in order to make `codegen_out_of_range(target_type : FloatType, arg_type : IntegerType, arg)` a total function. But there is no check needed if converting from Int to Float64 (or any Int but UInt128 to Float32), so is to emit less code.

Would you prefer to make `codegen_raise_overflow_cond` check if the `overflow` argument is a literal false value and avoid emitting code?